ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/datetime/ParseTime.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.datetime;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import io.confluent.ksql.function.FunctionCategory;
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.util.KsqlConstants;
+import java.sql.Time;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.ExecutionException;
+
+@UdfDescription(
+    name = "parse_time",
+    category = FunctionCategory.DATE_TIME,
+    author = KsqlConstants.CONFLUENT_AUTHOR,
+    description = "Converts a string representation of a time in the given format"
+        + " into a TIME value."
+)
+public class ParseTime {
+
+  private final LoadingCache<String, DateTimeFormatter> formatters =
+      CacheBuilder.newBuilder()
+          .maximumSize(1000)
+          .build(CacheLoader.from(DateTimeFormatter::ofPattern));
+
+  @Udf(description = "Converts a string representation of a time in the given format"
+      + " into the TIME value.")
+  public Time parseTime(
+      @UdfParameter(
+          description = "The string representation of a time.") final String formattedTime,
+      @UdfParameter(
+          description = "The format pattern should be in the format expected by"
+              + " java.time.format.DateTimeFormatter.") final String formatPattern) {
+    try {
+      final DateTimeFormatter formatter = formatters.get(formatPattern);
+      return new Time(LocalTime.parse(formattedTime, formatter).toNanoOfDay() / 1000000);
```

Comment:
Same two questions from FormatTime.
* Do we want to allow date characters in the format? I don't think we shoud.
* Can we use a constant variable for the nano per second value?

Reply:
Currently, if we parse something like `parse_time('2021 05:45', 'yyyy HH:mm)`, then it will parse everything but only return the time component (so in this case, it returns 05:45). It's weird that Local time.parse doesn't throw anything.

Reply:
Added a check to reject formats with non-time elements